bot/exts/filters/filtering.py

```diff
@@ -478,15 +478,16 @@ async def _has_urls(self, text: str) -> Tuple[bool, Optional[str]]:
         Second return value is a reason of URL blacklisting (can be None).
         """
         text = self.clean_input(text)
-        if not URL_RE.search(text):
+        matches = URL_RE.findall(text)
+        if not matches:
```

Comment:
You could simplify the code a bit like this:
```py
        domain_blacklist = self._get_filterlist_items("domain_name", allowed=False)
        for match in URL_RE.finditer(text):
            for url in domain_blacklist:
                if url.lower() in match.lower():
                    return True, self._get_filterlist_value("domain_name", url, allowed=False)["comment"]
```
Then you don't need the `if` check and a new local variable.

Reply:
Btw, why return a flag and an `Optional[str]` when you can just return `Optional[str]`? That would also play better with type checkers.

Reply:
Refactored that an pushed it, thanks :D

It seems all of the filters are expected to return a tuple like this. So changing that would require a larger change. As far as I know, #1530 plans to change how to do filtering quite a lot.

Reply:
oh i see